BWT/BWT.cs

```diff
@@ -0,0 +1,104 @@
+public static class BWT
+{
+    public static (string, int) Transform(string str)
+    {
+        var BWTString = new System.Text.StringBuilder();
+        var sortedShifts = SuffixArrayBuild(str);
+        var originalStringIndex = 1;
+        for (var i = 0; i < str.Length; ++i)
+        {
+            if (sortedShifts[i] == 0)
+            {
+                originalStringIndex = i + 1;
+                BWTString.Append(str[^1]);
+            }
+            else
+            {
+                BWTString.Append(str[sortedShifts[i] - 1]);
+            }
+        }
+        return (BWTString.ToString(), originalStringIndex);
+    }
+
+    public static string ReverseTransform(string BWTString, int originalStringIndex)
+    {   
+        if (BWTString.Length == 0 && originalStringIndex == 1)
+        {
+            return string.Empty;
+        }
+        if (originalStringIndex <= 0 || originalStringIndex > BWTString.Length)
+        {
+            Console.WriteLine("Incorrect input");
+            return string.Empty;
```

Comment:
С прошлого ревью не исправлено

```suggestion
            throw new ArgumentOutOfRangeException(paramName: nameof(originalStringIndex));
```